### ISSUE TITLE
ruby-lint-v2.0.4 produces results based on the order of input ruby files

### ISSUE THREAD

**Issue by vamshikr** [YorickPeterse/ruby-lint#148]
I am trying to use *ruby-lint-v2.0.4* on *ruby-v2.2.2* installation on *Scientific Linux* plaftorm. The problem I am seeing is ruby-lint is reporting different number of results based on the order of the files passed are arguments to the command.

Example: `ruby-lint file1.rb file2.rb` produces a different report than `ruby-lint file2.rb file1.rb`

When I run ruby-lint on ruby source files (all the *.rb* files in *lib* directory) in **capistrano**  (https://github.com/capistrano/capistrano), ruby-lint reports different results based on the order or files passed on the comand line and the number of issues reported varies from **1** to **700+** .

Why should the order matter and why am I seeing such a big variation?

**Comment by YorickPeterse** [comment]
What exactly are you running? Running `ruby-lint lib` on the Capistrano repository only results in the following output:

```
version.rb: warning: line 2, column 3: unused constant VERSION
upload_task.rb: error: line 4, column 22: undefined constant Rake::FileCreationTask
upload_task.rb: error: line 4, column 22: undefined constant Rake
svn.rb: warning: line 12, column 21: `*' interpreted as argument prefix
```

**Comment by vamshikr** [comment]
Thanks for the quick response, appreciate it.

Here are the list of commands that I ran:
```sh
% gem fetch capistrano
% gem unpack capistrano
% cd capistrano-3.4.0
% find lib/ -name '*.rb' > rubyfiles.txt
% for i in $(seq 1 10); do ruby-lint $(cat rubyfiles.txt | shuf | tr '\n' ' ') | wc -l; done
```
The number of weaknesses reported on my machine for the 10 runs are "57, 2, 53, 72, 1, 1, 22, 12, 56, 11"

**Comment by YorickPeterse** [comment]
This is caused by the following code: https://github.com/YorickPeterse/ruby-lint/blob/f1c3aa396c815b42524cfaab5e2abdd74d5bd081/lib/ruby-lint/runner.rb#L34-L38

Apparently one of the files triggers the following warning:

    svn.rb: warning: line 12, column 21: `*' interpreted as argument prefix

This then triggers the `return` above, skipping any following files. Sadly I don't think there's much I can do about this as the pure Ruby version of Racc still suffers from the problem mentioned in the source comment if I'm not mistaken.

**Comment by mvidner** [comment]
This one can be closed because it is fixed by https://github.com/YorickPeterse/ruby-lint/pull/177.
